isthmus/src/main/java/io/substrait/isthmus/SubstraitRelNodeConverter.java

```diff
@@ -629,24 +629,53 @@ public RelNode visit(VirtualTableScan virtualTableScan, Context context) {
         typeFactory.createStructType(fieldTypes, correctFieldNames);
 
     final List<ImmutableList<RexLiteral>> tuples = new ArrayList<>();
-    for (final Expression.StructLiteral row : virtualTableScan.getRows()) {
+    final List<RexNode> projectExpressions = new ArrayList<>();
+
+    for (final Expression rowExpr : virtualTableScan.getRows()) {
```

Comment:
You can use a tighter type bound here.
```suggestion
    for (final Expression.StructNested rowExpr : virtualTableScan.getRows()) {
```